src/hooks/loadables/useLoadBalances.ts

```diff
@@ -27,10 +27,16 @@ export const useTokenListSetting = (): boolean | undefined => {
   return isTrustedTokenList
 }
 
-const mergeBalances = (balance1: SafeBalanceResponse, balance2: SafeBalanceResponse): SafeBalanceResponse => {
+const mergeBalances = (cgw: SafeBalanceResponse, sn: SafeBalanceResponse): SafeBalanceResponse => {
+  // Create a Map using token addresses as keys
+  const uniqueBalances = new Map(
+    // Process SafeNet items first so they take precedence
+    [...sn.items, ...cgw.items].map((item) => [item.tokenInfo.address, item]),
+  )
+
   return {
-    fiatTotal: balance1.fiatTotal + balance2.fiatTotal,
-    items: [...balance1.items, ...balance2.items],
+    fiatTotal: cgw.fiatTotal + sn.fiatTotal,
```

Comment:
Won't the `fiatTotal` be too high if balance items occurred in both responses as they would be counted twice?

Reply:
yes it would, but currently, safenet is not returning any fiat balance, so it always remains the cgw fiat balance.

just FYI, this is a throwaway code, for production we expect this to be handled on the backend